adobecom/milo
# [MWPW-159181] - links conversion should handle extensions

## Description
There have been multiple requests for the link conversion feature to handle extensions.
The extensions will be removed by default for `.page` or `.live` urls.

## Related Issue
Resolves: [MWPW-159181](https://jira.corp.adobe.com/browse/MWPW-159181)

## Testing instructions
Using the **[test after URL](https://mwpw-159181-html-extension--milo--robert-bogos.hlx.page/drafts/rbogos/env-relative-urls)**, check if the `link authored to send to https://www.adobe.com/acrobat/free-trial-download.html` link is converted to point to `https://main--dc--adobecom.hlx.page/acrobat/free-trial-download` which has the `.html` extension removed.

## Test URLs
**Acrobat:**
- Before: https://www.stage.adobe.com/acrobat.html
- After: https://www.stage.adobe.com/acrobat.html?milolibs=mwpw-159181-html-extension--milo--robert-bogos

**BACOM:**
- Before: https://main--bacom--adobecom.hlx.page/products/real-time-customer-data-platform/rtcdp
- After: https://main--bacom--adobecom.hlx.page/products/real-time-customer-data-platform/rtcdp?milolibs=mwpw-159181-html-extension--milo--robert-bogos

**CC:**
- Before: https://main--cc--adobecom.hlx.page/drafts/rbogos/default?martech=off
- After: https://main--cc--adobecom.hlx.page/drafts/rbogos/default?martech=off&milolibs=mwpw-159181-html-extension--milo--robert-bogos

**Homepage:**
- Before: https://main--homepage--adobecom.hlx.page/homepage/index-loggedout
- After: https://main--homepage--adobecom.hlx.page/homepage/index-loggedout?milolibs=mwpw-159181-html-extension--milo--robert-bogos

**Blog:**
- Before: https://main--blog--adobecom.hlx.page/
- After: https://main--blog--adobecom.hlx.page/?milolibs=mwpw-159181-html-extension--milo--robert-bogos

**Milo:**
- Before: https://main--milo--adobecom.hlx.page/drafts/rbogos/env-relative-urls?martech=off
- After: https://mwpw-159181-html-extension--milo--robert-bogos.hlx.page/drafts/rbogos/env-relative-urls?martech=off
